### ISSUE TITLE
PinInput required attribute not added to input elements

### ISSUE THREAD

**Issue by henrymcl** [mantinedev/mantine#6588]
### Dependencies check up

- [X] I have verified that I use latest version of all @mantine/* packages

### What version of @mantine/* packages do you have in package.json?

```json
{
		"@mantine/core": "^6.0.19",
		"@mantine/dates": "^6.0.21",
		"@mantine/dropzone": "^6.0.21",
		"@mantine/form": "^6.0.21",
		"@mantine/hooks": "^6.0.19",
		"@mantine/notifications": "^6.0.21"
}
```
**package-lock.json:**
```json
{
		"node_modules/@mantine/core": {
			"version": "6.0.21"
		},
		"node_modules/@mantine/dates": {
			"version": "6.0.21"
		},
		"node_modules/@mantine/dropzone": {
			"version": "6.0.21"
		},
		"node_modules/@mantine/form": {
			"version": "6.0.21"
		},
		"node_modules/@mantine/hooks": {
			"version": "6.0.21"
		},
		"node_modules/@mantine/notifications": {
			"version": "6.0.21"
		}
}
```

### What package has an issue?

@mantine/core

### What framework do you use?

Vite

### In which browsers you can reproduce the issue?

Chrome

### Describe the bug

PinInput required attribute is not added to input elements, instead it's added to the parent `.mantine-PinInput-root` element.

I'm actually using 6.x.x but 7.x.x has it too.

### If possible, include a link to a codesandbox with a minimal reproduction

https://codesandbox.io/p/sandbox/pensive-haze-k5n3n3?file=%2Fsrc%2FApp.tsx

### Self-service

- [ ] I would be willing to implement a fix for this issue
*I'm stuck at the `yarn` step so I can't implement a fix*

**Comment by henrymcl** [comment]
I noticed that the `required` prop is no longer available in version 7, yet it's really helpful for HTML5 based form validation.

**Comment by rtivital** [comment]
Will be available with next patch:

```tsx
<PinInput getInputProps={() => ({ required: true })} />
```

**Comment by rtivital** [comment]
Fixed in [7.12.1](https://github.com/mantinedev/mantine/releases/tag/7.12.1)
